File: src/main/java/de/pbauerochse/worklogviewer/youtrack/ProgressCallback.java
package de.pbauerochse.worklogviewer.youtrack;

/**
 * Interface that allows updating the progress
 * of a certain task
 */
public interface ProgressCallback {

    void incrementProgress(String message, int amount);

    default void incrementProgress(String message) {
        incrementProgress(message, 10);
    }

}

File: api/src/main/java/de/pbauerochse/worklogviewer/connector/ProgressCallback.kt
package de.pbauerochse.worklogviewer.connector

/**
 * Interface that allows updating the progress
 * of a certain task
 */
interface ProgressCallback {

    /**
     * Update the current progress
     * @param message the message to display
     * @param amount the amount of progress in percent
     */
    fun setProgress(message: String, amount: Double)

    fun setProgress(message: String, amount: Int) {
        setProgress(message, amount.toDouble())
    }

}
